### PR TITLE
cppunit: add cppunit/1.15.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **cppunit/1.15.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cppunit/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+from contextlib import contextmanager
+import os
+
+
+class CppunitConan(ConanFile):
+    name = "cppunit"
+    description = "UNKNOWN DESCRIPTION"
+    topics = "UNKNOWN TOPICS"
+    license = "UNKNOWN LICENSE"
+    homepage = "UNKNOWN HOMEPAGE"
```
Comment (intelligide) [inline recipes/cppunit/all/conanfile.py:11]:
```suggestion
    description = "CppUnit is the C++ port of the famous JUnit framework for unit testing. Test output is in XML for automatic testing and GUI based for supervised tests."
    topics = ("conan", "cppunit", "unit-test", "tdd")
    license = " LGPL-2.1-or-later"
    homepage = "https://freedesktop.org/wiki/Software/cppunit/"
```

File: recipes/cppunit/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libcppunit.la"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["cppunit"]
```
Comment (intelligide) [inline recipes/cppunit/all/conanfile.py:101]: It produces `cppunit.dll.lib` on Windows

File: recipes/cppunit/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/cppunit/all/conanfile.py:11]: Thanks, I forgot to update the template.

File: recipes/cppunit/all/conanfile.py (hunk 2 above)
Comment (madebr) [inline recipes/cppunit/all/conanfile.py:101]:
```suggestion
        libsuffix = ".dll" if self.settings.os == "Linux" and self.options.shared else ""
        self.cpp_info.libs = ["cppunit" + libsuffix]
```

Comment (madebr) [reply recipes/cppunit/all/conanfile.py]: Thanks I've updated the name.

File: recipes/cppunit/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        os.unlink(os.path.join(self.package_folder, "lib", "libcppunit.la"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        libsuffix = ".dll" if self.settings.os == "Linux" and self.options.shared else ""
+        self.cpp_info.libs = ["cppunit" + libsuffix]
+        if not self.options.shared:
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs = ["dl"]
```
Comment (madebr) [inline recipes/cppunit/all/conanfile.py:105]:
```suggestion
                self.cpp_info.system_libs.append("dl")
```

File: recipes/cppunit/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libcppunit.la"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        libsuffix = ".dll" if self.settings.os == "Linux" and self.options.shared else ""
+        self.cpp_info.libs = ["cppunit" + libsuffix]
+        if not self.options.shared:
+            if self.settings.os == "Linux":
```
Comment (madebr) [inline recipes/cppunit/all/conanfile.py:104]:
```suggestion
            stdlib = tools.stdcpp_library(self)
            if stdlib:
                self.cpp_info.system_libs.append(stdlib)
            if self.settings.os == "Linux":
```

File: recipes/cppunit/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libcppunit.la"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        libsuffix = ".dll" if self.settings.os == "Linux" and self.options.shared else ""
```
Comment (madebr) [inline recipes/cppunit/all/conanfile.py:101]:
```suggestion
        libsuffix = ".dll" if self.settings.os == "Windows" and self.options.shared else ""
```

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by madebr** [comment]
@SpaceIm @prince-chrismc 
Can (one of) you please review this?

File: recipes/cppunit/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libcppunit.la"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        libsuffix = (".dll" if self.settings.os == "Windows" and self.options.shared else "") + (".lib" if self.settings.compiler == "Visual Studio" else "")
```
Comment (prince-chrismc) [inline recipes/cppunit/all/conanfile.py:102]: with MSVC and shared `libsuffix =".dll.lib` is that correct?

File: recipes/cppunit/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        self.cpp_info.libs = ["cppunit" + libsuffix]
+        if not self.options.shared:
+            stdlib = tools.stdcpp_library(self)
+            if stdlib:
+                self.cpp_info.system_libs.append(stdlib)
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs.append("dl")
+        if self.options.shared:
+            if self.settings.os == "Windows":
+                self.cpp_info.defines.append("CPPUNIT_DLL")
```
Comment (prince-chrismc) [inline recipes/cppunit/all/conanfile.py:112]:
```suggestion
        if self.options.shared and self.settings.os == "Windows":
                self.cpp_info.defines.append("CPPUNIT_DLL")
```

File: recipes/cppunit/all/conanfile.py (hunk 6 above)
Comment (madebr) [reply recipes/cppunit/all/conanfile.py:102]: Yes, I tested this on MSVC.
If I remember correct, in general, when libtool is used to build shared libraries,
it generates a `.dll.lib` file for MSVC and a `.dll.a` file for mingw.

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/cppunit/all/conanfile.py (hunk 6 above)
Comment (SpaceIm) [reply recipes/cppunit/all/conanfile.py:102]: I suggest to rename import lib suffix if visual studio.

Comment (SpaceIm) [reply recipes/cppunit/all/conanfile.py:102]: I would advice to rename import lib suffix if visual studio (.dll.lib => .lib)

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**